server/utils/checkYourAnswersUtils.ts

```diff
@@ -293,3 +298,33 @@ export const getApplicantDetails = (application: Application | Cas2SubmittedAppl
     },
   ]
 }
+
+const findMatchedKeys = (obj: any, keysToFind: string[]): string[] => {
+  const keys: string[] = []
+
+  if (typeof obj === 'object' && obj !== null) {
+    for (const key in obj) {
+      if (keysToFind.includes(key)) {
+        keys.push(key)
+      }
+
+      keys.push(...findMatchedKeys(obj[key], keysToFind))
+    }
+  }
+
+  return keys
+}
+
+const processFindMatchedKeys = async (jsonBlob: any, keysToFind: string[]): Promise<string[]> => {
+  const keys = Object.keys(jsonBlob)
+  const result = await Promise.all(keys.map(key => findMatchedKeys(jsonBlob[key], keysToFind)))
+  return result.flat()
+}
+
+const filterKeysWithoutQuestions = async (questions: any, task: string, pageKeys: string[]): string[] => {
+  const permanentPagesWithoutQuestions = ['summary-data', 'oasys-import']
+  const dynamicPagesKeys = pageKeys.filter(item => !permanentPagesWithoutQuestions.includes(item))
+  const matchedKeys = await processFindMatchedKeys(questions[task], dynamicPagesKeys)
```

Comment:
We fan out across the list of keys here and each task will drill down and compare keys. Rather than going through each key and sub key sequentially.